addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -2304,6 +2304,76 @@ var FieldRadio = FieldSelection.extend({
     },
 });
 
+
+var FieldSelectionBadge = FieldSelection.extend({
+    template: null,
+    className: 'o_field_selection_badge',
+    tagName: 'span',
+    specialData: "_fetchSpecialMany2ones",
+    events: _.extend({}, AbstractField.prototype.events, {
+        'click span.o_selection_badge': '_onBadgeClicked',
+    }),
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @override
+     */
+    _renderEdit: function () {
+        var self = this;
+        this.currentValue = this.value;
+
+        if (this.field.type === 'many2one') {
+            this.currentValue = this.value && this.value.data.id;
+        }
+        this.$el.empty();
+        this.$el.html(qweb.render('FieldSelectionBadge', {'values': this.values, 'current_value': this.currentValue}));
+    },
+    /**
+     * Sets the possible field values. If the field is a many2one, those values
+     * may change during the lifecycle of the widget if the domain change (an
+     * onchange may change the domain).
+     *
+     * @private
+     * @override
+     */
+    _setValues: function () {
```

Comment:
Linting: according to https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines there should not be spaces between function and arguments (to propagate to other occurences)

Reply:
@tde-banana-odoo https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines#documentation
For space between function and arguments

Reply:
Oh yes, they don't use space in function myFunction(a, b) { /*...*/ } but well in _setValues: function (). Seems right then.

Reply:
the idea is that function is a javascript keyword